### DISCUSSION TITLE
Big or small problem:  Broken libmagic installation detected.

### DISCUSSION THREAD

**Discussion by avishavish1999** [LAAC-LSCP/ChildProject#352] [General]
after running commands overviewing the childproject, I got:

"Broken libmagic installation detected. The python-magic module is installed but can't be imported. Please check that both python-magic and the libmagic shared library are installed correctly. Uploading media other than images may not work." 

wondering what I can do to fix it or whether this is small or big.

**Comment by lucasgautheron** [comment]
Hello,

Please do:

```bash
pip uninstall ChildProject
pip install git+https://github.com/LAAC-LSCP/ChildProject.git
```

Note that this issue seems to be caused by the Zooniverse (panoptes) library. You won't be able to run the zooniverse pipeline until your installation of python-magic is fixed.

I've updated the package so that it won't load Zooniverse's python library if not necessary.

**Reply by alecristia** [reply]
thanks! Avishai is on windows, but I think your suggestion should still work.
He doesn't need to use zooniverse - so would you recommend then ignoring the message?

**Reply by lucasgautheron** [reply]
Yes, I think this message can be ignored!

**Reply by alecristia** [reply]
Note, problem also observed in mac:
> I attempted to install the ChildProject package yesterday, following the set up instructions for Mac OS on my Macbook. I had my partner (who is a software engineer) assist me with the process. We installed brew, however git-annex was not recognised as a valid package name. I was able to set up and activate the conda environment, but running childproject --help resulted in the following error: 

```(childproject) jeanheng@Jeans-MBP childproject % child-project --help
Broken libmagic installation detected. The python-magic module is installed but can't be imported. Please check that both python-magic and the libmagic shared library are installed correctly. Uploading media other than images may not work.
```

**Reply by alecristia** [reply]
RE the git-annex problem alluded to above:
You absolutely need git-annex to work. There are a few ways of installing it:
- Can you please double check the spelling? `brew install git-annex` should have worked. I'm pretty certain it exists: https://git-annex.branchable.com/install/OSX/Homebrew/
- Although I've never tried it, you could perhaps try to install the app at the top of: https://git-annex.branchable.com/install/OSX/

**Reply by lucasgautheron** [reply]
> Note, problem also observed in mac:
> 
> > I attempted to install the ChildProject package yesterday, following the set up instructions for Mac OS on my Macbook. I had my partner (who is a software engineer) assist me with the process. We installed brew, however git-annex was not recognised as a valid package name. I was able to set up and activate the conda environment, but running childproject --help resulted in the following error:
> 
> ```
> Broken libmagic installation detected. The python-magic module is installed but can't be imported. Please check that both python-magic and the libmagic shared library are installed correctly. Uploading media other than images may not work.
> ```

Hello! I know I promised I would turn off the notifications.. But if it helps: https://github.com/LAAC-LSCP/ChildProject/commit/f3486cec56599f1a8b6faea761543b234dbe27ea > the python-magic message should not appear anymore if the package is installed from the master branch.  However the underlying issue will need to be fixed if you plan to use the zooniverse upload-chunks pipeline. I recommend asking zooniverse's people about that!

Cheers

**Comment by kalenkovich** [comment]
For those getting the libmagic warning on Mac: you can install it with

```
conda install -c conda-forge libmagic
```
